File: commandapi-platforms/commandapi-paper/commandapi-paper-test/commandapi-paper-test-tests/src/test/java/dev/jorel/commandapi/test/Mut.java
package dev.jorel.commandapi.test;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.NoSuchElementException;
import java.util.Optional;

/**
 * A mutable wrapper of an object, as a list
 *
 * @param <T>
 */
public class Mut<T> {
	
	public static <T> Mut<T> of() {
		return new Mut<T>();
	}

	public static <T> Mut<T> of(T obj) {
		return new Mut<T>(obj);
	}
	
	private final Deque<Optional<T>> value;
	
	private Mut() {
		value = new ArrayDeque<>();
	}
	
	private Mut(T obj) {
		this();
		set(obj);
	}
	
	public void set(T obj) {
		this.value.add(Optional.ofNullable(obj));
	}
	
	public T get() {
		if(this.value.size() == 0) {
			throw new NoSuchElementException();
		} else {
			return this.value.remove().orElse(null);
		}
	}
	
	@Override
	public String toString() {
		return String.valueOf(value);
	}
	
}

File: commandapi-platforms/commandapi-bukkit/commandapi-bukkit-test/commandapi-bukkit-kotlin-test/src/test/kotlin/dev/jorel/commandapi/test/Mut.kt
package dev.jorel.commandapi.test

import java.util.*

/**
 * A mutable wrapper of an object, as a list
 *
 * @param <T>
 */
class Mut<T : Any> private constructor() {
	private val value: Deque<Optional<T>>

	init {
		value = ArrayDeque()
	}

	private constructor(obj: T) : this() {
		set(obj)
	}

	fun set(obj: T) {
		value.add(Optional.ofNullable(obj))
	}

	fun get(): T {
		return if (value.size == 0) {
			throw NoSuchElementException()
		} else {
			value.remove().orElse(null)
		}
	}

	override fun toString(): String {
		return value.toString()
	}

	companion object {
		fun <T : Any> of(): Mut<T> {
			return Mut()
		}

		fun <T : Any> of(obj: T): Mut<T> {
			return Mut(obj)
		}
	}
}